compiler/rustc_const_eval/src/interpret/visitor.rs

```diff
@@ -131,6 +137,15 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> Value<'mir, 'tcx, M> for OpTy<'tc
     ) -> InterpResult<'tcx, Self> {
         ecx.operand_field(self, field)
     }
+
+    #[inline(always)]
+    fn strip_pat_ty(mut self) -> Self {
+        let ty::Pat(raw_ptr_ty, _) = *self.layout.ty.kind() else {
+            bug!("NonNull must contain a pattern type, but had {}", self.layout.ty)
```

Comment:
These assertions are confusing. The method name and doc comment to not talk about NonNull in any way.